amgut/lib/data_access/ag_data_access.py

```diff
@@ -428,14 +441,16 @@ def getConsent(self, survey_id):
             if result:
                 return {k: v for k, v in zip(colnames, result)}
 
-    def insertAGMultiple(self, ag_login_id, participant_name, field_name,
-                         field_value):
-        sql = ("insert into ag_survey_multiples (ag_login_id, "
-               "participant_name,item_name, item_value) values ('{0}','{1}',"
-               " '{2}', '{3}')").format(ag_login_id, participant_name,
-                                        field_name, field_value)
-        self.get_cursor().execute(sql)
-        self.connection.commit()
+    # NOTE: This method is vulnerable to SQL injection as-written but is not
```

Comment:
Yes please, thanks!

Flagging @adamrp just to make sure he saw this note re: animal survey. We
haven't redone it yet, so it's still carrying some fluff.

On Fri, Jan 2, 2015 at 8:22 PM, Beau Gunderson <notifications@github.com>
wrote:

> In amgut/lib/data_access/ag_data_access.py
> <https://github.com/biocore/american-gut-web/pull/354#discussion-diff-22429153>
> :
>
> > @@ -428,14 +441,16 @@ def getConsent(self, survey_id):
> >              if result:
> >                  return {k: v for k, v in zip(colnames, result)}
> >
> > -    def insertAGMultiple(self, ag_login_id, participant_name, field_name,
> > -                         field_value):
> > -        sql = ("insert into ag_survey_multiples (ag_login_id, "
> > -               "participant_name,item_name, item_value) values ('{0}','{1}',"
> > -               " '{2}', '{3}')").format(ag_login_id, participant_name,
> > -                                        field_name, field_value)
> > -        self.get_cursor().execute(sql)
> > -        self.connection.commit()
> > +    # NOTE: This method is vulnerable to SQL injection as-written but is not
>
> The one place where it was in use is inside the animal survey, but the
> bulk of that survey is commented out (including the line that references
> this function); I figured it may be used again at some point from there but
> wanted to note that potential for SQL injection so that it could be fixed
> before it was used again. Should it just be removed?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/biocore/american-gut-web/pull/354/files#r22429153>.
>